Writerside/topics/Customizing-Account-Login-and-Linking.md

```diff
@@ -0,0 +1,271 @@
+
+```Typescript
+  const providers = await getEnabledProviders();
+
+  const enabledProviders = providers.enabledProviders.map((providerString) => {
+    return LoginFormProvider[providerString] as LoginFormProvider
+  })
+```
+
+We then pass this data, along with a `targetURL` into the `LoginForm` component. The target url tells the LoginForm where to 
+redirect the user after they have successfully logged into the platform.
+
+```Typescript
+<LoginForm
+  enabledProviders={enabledProviders}
+  targetURL={`${catena_dashboard_url}/dashboard/accounts`}
+></LoginForm>
+```
+
+This means to customize the page itself you can add code to either `page.tsx` or to the `layout.tsx` in the same directory, however changing the form itself should be in the `LoginForm` component.
+
+## LinkPage
+
+The link page looks very similar to the login page, that is because it uses the same form component, this ensures we are consistent in how we apply styles across the platform.
+
+Visit `/link` in order to view the link page on your dashboard. You should see a screen like the one below:
+
+![linkPage.png](linkPage.png)
+
+The code for the link page is located at `/src/app/link/page.tsx` and should look very similar to the login page.
+
+The link page's code will look mostly similar to the login page, however the target url and layout is a bit different:
+
+```Typescript
+<div className={"w-full h-96 flex justify-center items-center"}>
+  <LoginForm targetURL={`${catena_dashboard_url}/link/step2?` + urlProviders.toString()} enabledProviders={enabledProviders}></LoginForm>
+</div>
+```
+
+Notice we are now sending the user to `/link/step2` once login is complete, we are also encoding the enabled providers as a url parameter so we do not have to re-fetch the enabled providers on the next page.
+
+If you log in with a provider you will be sent to the `/link/step2` route which will allow you to complete the second part of the linking process.
+
+## LinkPage Step 2
+
+The code for this page is at `/src/app/link/step2/page.tsx`. This page allows the user to complete the linking flow by logging in with the provider they want to link.
+
+The example below shows what a user may see if they have linked google and unsafe providers to the same account.
+
+![linkPageStep2.png](linkPageStep2.png)
+
+This page must juggle a few session tokens in order to complete the login step. It will do the following:
+
+1. Complete the user's login flow 
+2. Persist the user's login session token in local storage. It must retrieve this from the Next.js backend which maintains the state for the user. This token will be used in the next step.
+```Typescript
+    const jsonData: { account: Account } = await getLoginInfo.json()
+    const getCatenaSessionID = await fetch(`/api/v1/authentication/session`, {
+      credentials: "include",
+    })
+    const sessionData: { session: string } = await getCatenaSessionID.json()
```

Comment:
Is this snippet supposed to show storing the session in local storage?

Reply:
Updating for clarity